src/Symfony/Component/HttpClient/DataCollector/HttpClientDataCollector.php

```diff
@@ -95,6 +95,17 @@ public function reset()
         ];
     }
 
+    /**
+     * Let's be defensive : we authorize only size of 4Mo for escapeshellarg() argument to avoid a fatal error
+     *
+     * @see https://github.com/php/php-src/blob/9458f5f2c8a8e3d6c65cc181747a5a75654b7c6e/ext/standard/exec.c#L397
+     * @see https://github.com/php/php-src/blob/9458f5f2c8a8e3d6c65cc181747a5a75654b7c6e/ext/standard/exec.c#77
+     */
+    private function argMaxLengthIsSafe(string $payload): bool
+    {
+        return strlen($payload) <= 4096;
```

Comment:
4096 bytes is super small, this must be a mistake since you mention 4Mo in the comment
please also move the method at the bottom of the class and add a leading slash to `\strlen`

Reply:
Looking at the PHP codebase, the size seems to be in bytes.

Reply:
fabbot (aka php-cs-fixer) is on vacation I think these days, it doesn't show up in the CI reports since a few weeks :P

Reply:
> Looking at the PHP codebase, the size seems to be in bytes.

OK but then we cannot go by this limit, it's way too low I think.

Reply:
It's difficult to be precise when the value is not accessible with PHP functions. What value do you think is enough ? Maybe 8 Mo to follow PHP algo :man_shrugging:

Reply:
@fmata where are you taking this 8Mo ? I don't see that in the PHP code. The hardcoded values I see in your links are 8KB on Windows and 4KB as fallback value. I don't see 8MB anywhere.

@nicolas-grekas if this size limit is too small, then we need to stop using `escapeshellargs`, because it is the one having a limit.

Reply:
On my machine, I can go up to `escapeshellarg(str_repeat('-', 2097149));`.
What about others?

Reply:
Well, as seen in the PHP source code, the value of `cmd_max_len` can be set in a bunch of different ways during compilation. So it can indeed be higher on some systems. But this is not exposed to userland.

anyway, disabling the copy as curl feature in the profile is better than doing a fatal error when running the HTTP request

Reply:
So apparently on Windows we can't go above 8kio, and on linux we need to set a reasonably portable limit. Here is a suggestion, going a bit below the limit to account for escaping if that matters?
`return  \strlen($payload) < ('\\' === \DIRECTORY_SEPARATOR ?8100 : 256000;`